drivers/flash/flash_stm32_qspi.c

```diff
@@ -133,17 +136,29 @@ static inline void qspi_set_address_size(const struct device *dev,
 	cmd->AddressSize = QSPI_ADDRESS_24_BITS;
 }
 
-static inline void qspi_prepare_quad_read(const struct device *dev,
+static inline int qspi_prepare_quad_read(const struct device *dev,
 					  QSPI_CommandTypeDef *cmd)
 {
 	struct flash_stm32_qspi_data *dev_data = dev->data;
 
 	if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO) && dev_data->flag_quad_io_en) {
+		switch (dev_data->mode) {
+		case JESD216_MODE_114:
+			cmd->AddressMode = QSPI_ADDRESS_1_LINE;
+			break;
+		case JESD216_MODE_144:
+			cmd->AddressMode = QSPI_ADDRESS_4_LINES;
+			break;
+		default:
+			return -ENOTSUP;
```

Comment:
Is that possible at this step ? If not, what about a mere if/else

Reply:
It is not possible to have a not supported value here as well as in the reading prepare function.

However, that helps to clearly communicate to another dev or a maintainer that nothing else is supported. This might help to detect errors when DTS `writeoc` enum is extended or when somebody adds other modes (potentially refactors to support dual/octal modes).